stdlib: escape strings of S-Expressions

zerbina opened the pull request:
## Summary
- use `strutils.escape` to escape the string when rendering an S-Expression via `toLines` and `treeRepr`
- make the S-Expression parser support `\x` escape sequences

Together, this allows the output of `treeRepr` and `toLines` to be parsed back into `SexpNode`s without losing information.

Fixes https://github.com/nim-works/nimskull/issues/263

---
<!-- Note: section break (`---`) onwards is not in CI merge commit -->

@saem Merging this PR can wait until #524 is merged.

<!--
Pull Request(PR) Help

Before Merge Ensure:
* title reads like a short changelog line entry
* code includes tests and is documented
* leave the source better than before, but split out big reformats

See contributor (guide)[https://nim-works.github.io/nimskull/contributing.html]
for details, especially if you're new to this project.

Tips that make PRs easier:
* for big/impactful changes, start with chat/discussions to refine ideas
* refine the pull request message over time; don't have to nail it in one go
* handle the single commit message requirement at the end of review

haxscramper reviewed (approved):
Looks good to me

saem reviewed (approved):
bors r+

alaviss commented:
I believe sexp was using JSON-style escape, is that compatible with strutils-based escape?

zerbina committed:
stdlib: always escape string S-Expressions

## Summary
- use JSON-style string escaping (`escapeJson`) when rendering an
  S-Expression to a text representation via `toLines` and `treeRepr`
- make the S-Expression parser support `\x` escape sequences
- add tests for `\x` and `\u` escape sequences

Together, this allows the output of `toLines` to be parsed back into
`SexpNode`s without losing information.

Fixes https://github.com/nim-works/nimskull/issues/263

zerbina commented:
> I believe sexp was using JSON-style escape, is that compatible with strutils-based escape?

I think not, as JSON-style escaping assumes UTF-8 encoding while `strutils.escape` does not.
Both `treeRepr` and `toLines` are now adjusted to use `escapeJson`, which is in line with `toPretty` (i.e. `$`).

saem commented:
Bors r+

g2g
